docsTech/processing.md

```diff
@@ -0,0 +1,263 @@
+# Processing
+
+## Introduction
+
+## Pipe-processing
+
+### List of pipe-processing
+
+#### `canBeReordered` (pro only)
+
+**Publisher**: `useGridColumnReorder` when dragging a column over another.
+
+**Why register to this processing**:
+
+**Example**:
+
+```ts
+// Column Pinning plugin
+const checkIfCanBeReordered = React.useCallback<GridPipeProcessor<'canBeReordered'>>(
+  (initialValue, { targetIndex }) => {
+    const [leftPinnedColumns, rightPinnedColumns] = filterColumns(
+      pinnedColumns,
+      gridVisibleColumnFieldsSelector(apiRef),
+    );
+
+    if (leftPinnedColumns.length === 0 && rightPinnedColumns.length === 0) {
+      return initialValue;
+    }
+
+    if (leftPinnedColumns.length > 0 && targetIndex < leftPinnedColumns.length) {
+      return false;
+    }
+
+    if (rightPinnedColumns.length > 0) {
+      const visibleColumns = gridVisibleColumnDefinitionsSelector(apiRef);
+      const firstRightPinnedColumnIndex = visibleColumns.length - rightPinnedColumns.length;
+      return targetIndex >= firstRightPinnedColumnIndex ? false : initialValue;
+    }
+
+    return initialValue;
+  },
+  [apiRef, pinnedColumns],
+);
+
+useGridRegisterPipeProcessor(apiRef, 'canBeReordered', checkIfCanBeReordered);
+```
+
+#### `columnMenu`
+
+**Publisher**: `GridColumnMenu` component on render.
+
+**Why register to this processing**: Add one or multiple menu items to `GridColumnMenu`.
+
+**Example**:
+
+```tsx
+// Column pinning plugin
```

Comment:
@alexfauquette what do you think is the best to demonstrate how each processor work ?

- A copy paste of a real use case, potentially simplified (I removed logic that were not interesting)
- A link to the file 
- A fake use case